tools/extra/hssi/hssistats.py

```diff
@@ -0,0 +1,171 @@
+                      ('rx_unicast', 10),
+                      ('rx_multicast', 11),
+                      ('rx_broadcast', 12),
+                      ('tx_discards', 13),
+                      ('tx_unicast', 14),
+                      ('tx_multicast', 15),
+                      ('tx_broadcast', 16),
+                      ('ether_drops', 18),
+                      ('rx_total_packets', 19),
+                      ('rx_undersize', 20),
+                      ('rx_oversize', 21),
+                      ('rx_64_bytes', 22),
+                      ('rx_65_127_bytes', 23),
+                      ('rx_128_255_bytes', 24),
+                      ('rx_256_511_bytes', 25),
+                      ('rx_512_1023_bytes', 26),
+                      ('rx_1024_1518_bytes', 27),
+                      ('rx_gte_1519_bytes', 28),
+                      ('rx_jabbers', 29),
+                      ('rx_runts', 30))
+
+    def __init__(self, args):
+        self.pcieaddress = args.pcieaddress
+        self.hssi_grps = args.hssi_grps
+        HSSICOMMON.__init__(self)
+
+    def get_hssi_stats(self):
+        """
+        clear ctl address and ctl sts CSR
+        write 0x3 value ctl address and address bit
+        write read cmd 0x1 value ctl sts csr
+        poll for status
+        read LSB stats
+        clear ctl address and ctl sts CSR
+        write 0x3 value ctl address , address bit  and set 31 bit
+        write read 0x1 value ctl sts csr
+        poll for status
+        read MSB stats
+        print stats
+        """
+        self.open(self.hssi_grps[0][0])
+        ctl_addr = hssi_ctl_addr(0)
+        ctl_addr.sal_cmd(HSSI_SALCMD.READ_MAC_STATISTIC.value)
+        value = 0
+        print("------------HSSI stats------------")
+        for str, reg in self.hssi_eth_stats:
+
+            # Read LSB value
+            ctl_addr.addressbit(reg)
+            value_lsb = self.read_reg(0, ctl_addr.value)
+
+            # Read MSB value
+            ctl_addr.value = self.register_field_set(ctl_addr.value, 31, 1, 1)
+            value_msb = self.read_reg(0, ctl_addr.value)
+
+            # 64 bit value
+            value = (value_msb << 32) | (value_lsb)
+
+            print("{0: <20}".format(str), end=' | ')
+            print("{0: >12}".format(value), end=' | ')
+            print()
+
+        self.close()
+
+    def hssi_stats_start(self):
+        """
+        print hssi info
+        get hssi stats
+        """
+        print("----hssi_stats_start----")
+        self.hssi_info(self.hssi_grps[0][0])
+        self.get_hssi_stats()
+
+
+def main():
+    """
+    parse input arguemnts pciaddress and mtu
+    enum fpga pcie devices and find match
+    read hssi statistics
+    """
+    parser = argparse.ArgumentParser()
+
+    pcieaddress_help = 'bdf of device to program \
+                        (e.g. 04:00.0 or 0000:04:00.0).' \
+                       ' Optional when one device in system.'
+    parser.add_argument('--pcieaddress', '-P',
```

Comment:
Could we use --pcie-address?

Reply:
changed input arguments to --pcie-address , fixed